runtime/server/auth/permissions.go

```diff
@@ -0,0 +1,19 @@
+package auth
+
+// Permission represents runtime access permissions.
+type Permission int
+
+const (
+	// System-level permissions
+	ManageInstances Permission = 0x01
+
+	// Instance-level permissions
+	ReadInstance  Permission = 0x11
+	EditInstance  Permission = 0x12
+	ReadRepo      Permission = 0x13
+	EditRepo      Permission = 0x14
+	ReadObjects   Permission = 0x15
+	ReadOLAP      Permission = 0x16
+	ReadMetrics   Permission = 0x17
+	ReadProfiling Permission = 0x18
```

Comment:
Where would we want to distinguish between these permissions, for example if user has `ReadMetrics` and `ReadProfiling` but not `ReadOLAP` what are we restricting here ?

Reply:
I should probably document these. My thinking was that `ReadOLAP` is for direct SQL access, whereas `ReadMetrics` is only for metrics/dashboard APIs, and `ReadProfiling` is only for model profiling APIs. Does that make sense?

E.g. we don't want to give public dashboard viewers direct OLAP SQL access

Reply:
I was just thinking if there is some natural grouping here. For example, for a user to see dashboard they will always need both `ReadMetrics` and `ReadProfiling`. I believe `ReadOLAP` will be used during modeling but then this is not sufficient to have working modeling view other permissions are also needed. One problem that I find with this approach is if tomorrow for rendering dashboard front wants to do a simple OLAP query then it require `ReadOLAP` permission and that will break our initial assumptions. 

Just thinking out loud here, another approach would be of have permission like `ReadSource` on `SourceName` or `*` (this will propagate to models and metrics view derived from this source). But may this is very broad as this would give user permission to run any random query on the source if they get access to the query API.

Reply:
My philosophy is that these permissions just map up to actual things you can do. Then it's on the control-plane side to map them up to use cases (the grouping). If we added the groupings here (e.g. by using a concept of hard-coded roles instead), then we would lose flexibility.

Regarding object-level permissions (like read access on a specific dashboard) – I think you're right that would make sense at some point. But at that granularity, the policies get very complicated, so I think it should be done as part of considering granular access policies defined in code (which should also cover row-based/column-based filters). So IMO we can skip on that for now and just have these "all or nothing" permissions.

Reply:
sounds good